### PR TITLE
Feature: Best Practices :fast_forward: 

### PR DESCRIPTION
**Pull request by AdornChoga**
In this pull request, we applied the following best practices:

- YAGNI, DRY, and KISS
- SOLID

### CODE REVIEW

File: modules/app_people.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  def create_student(age, name, people)
+    print 'Has parent permission to? [Y/N]: '
+    permission = gets.chomp
+    yes_answers = %w[y yes]
+    no_answers = %w[n no]
+    if yes_answers.include? permission.downcase
+      student = Student.new(age, name, true)
+    elsif no_answers.include? permission.downcase
+      student = Student.new(age, name, false)
+    else
```
Comment (mikethreels) [inline modules/app_people.rb]: - [ ] in this statement you are repeating the same code twice this is not according to [the DRY principle ](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid)please change it so complies with the DRY rules.
- [ ] in this method you are printing to the console, getting data from the user and saving the data as well, this is not in line with the single responsibility rules, please have another look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more details.

File: modules/app_people.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    puts 'Person created successfully'
+  end
+
+  def create_teacher(age, name, people)
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(age, name, specialization)
+    people << teacher
+    puts 'Person created successfully'
+  end
```
Comment (mikethreels) [inline modules/app_people.rb]: - [ ] in this method you are printing to the console, getting data from the user and saving the data as well, this is not in line with the single responsibility rules, please have another look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more details.

File: modules/app_books.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  def create_book(books)
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    books << book
+    puts 'Book created successfully'
+  end
```
Comment (mikethreels) [inline modules/app_books.rb]: - [ ] in this method you are printing to the console, getting data from the user and saving the data as well, this is not in line with the single responsibility rules, please have another look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more details.

File: modules/app_rentals.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_index = gets.chomp.to_i
+    puts ''
+
+    print 'Date: '
+    date = gets.chomp
+
+    Rental.new(people[person_index], books[book_index], date)
+    puts 'Rental Created successfully'
```
Comment (mikethreels) [inline modules/app_rentals.rb]: - [ ] in this method you are printing to the console, getting data from the user and saving the data as well, this is not in line with the single responsibility rules, please have another look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more details.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Great job so far, I really like the way you're putting in extra effort to catch edge cases 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/app_people.rb (hunk 2 above)
Comment (AdornChoga) [reply modules/app_people.rb]: Hi, @mikethreels.
Thank you for the feedback. Do you mean that we should create a separate method for printing one line of code #28 and two more methods for the other parts of this create_teacher() method?
From our perspective, the create_teacher method is doing one job here, which is to create a teacher object by performing some micro-tasks.

File: main.rb [hunk 5]
```diff
@@ -4,21 +4,19 @@
```
Comment (Meltrust) [inline main.rb:4]: - [ ] OPTIONAL:  The SOLID can be improved, I'll explain:

## Explanation

Every class should have a single responsibility and methods that relate to the main functionality of that class.  Also, the user input, the "puts", and the interface should be apart from the logic.  

## Hints

- All requests for input and input validation should be placed in the `main.rb` file and/or classes/methods whose responsibilities are to request or validate input.  That way, we comply with the single responsibility rule.

- Also, in this case, the `app.rb` files and modules, should just serve as "listers" to populate the screen with data that the user requests (like show all books, etc).

That way, if you add more functionality in the future, you won't need to refactor the whole thing, thus making the app more easily scalable. 💪

**Review by Meltrust — APPROVED** [review]
Hi @AdornChoga,


## Wow, you did it 🎉
Brilliant
## Thank you for separating your input requests to the user, as requested by the previous reviewer 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/app_people.rb (hunk 2 above)
Comment (Meltrust) [reply modules/app_people.rb]: Hi @AdornChoga, I left you a short explanation on the review!

Comment (AdornChoga) [reply modules/app_people.rb]: Thanks a lot for the clarification @Meltrust.

**Pull request merged**